crates/cadenza-eval/src/eval.rs

```diff
@@ -1270,6 +929,524 @@ pub fn builtin_measure() -> BuiltinMacro {
     }
 }
 
+// =============================================================================
+// Built-in operators (registered as functions in the standard environment)
+// =============================================================================
+
+/// Creates the `+` addition operator.
+pub fn builtin_add() -> BuiltinFn {
+    BuiltinFn {
+        name: "+",
+        signature: Type::function(
```

Comment:
Wouldn't we have a union at the outer layer and not arguments? As in either the function takes/returns floats or imts. This type signature indicates something else